scenario/reload.go

```diff
@@ -101,8 +106,12 @@ func (settings ReloadSettings) Execute(sessionState *session.State,
 				// Get the progress using the request id we reserved for the reload
 				var progress *enigma.ProgressData
 				getProgress := func(ctx context.Context) error {
+					uplink := sessionState.Connection.Sense()
```

Comment:
still should check sessionState.Connection != here as this is in separate go-routine and something else could nil the connection.

Reply:
ahh, then getting an uplink from sessionState should be atomic or we should just accept the panics